src/crystal/system/event_loop/file_descriptor.cr

```diff
@@ -19,5 +19,13 @@ abstract class Crystal::EventLoop
 
     # Closes the file descriptor resource.
     abstract def close(file_descriptor : Crystal::System::FileDescriptor) : Nil
+
+    # Removes the file descriptor from the event loop. Can be used to free up
+    # memory resources associated with the file descriptor, as well as removing
+    # the file descriptor from kernel data structures.
+    #
+    # Called by `::IO::FileDescriptor#finalize` before closing the file
+    # descriptor. Errors shall be silently ignored.
+    abstract def delete(file_descriptor : Crystal::System::FileDescriptor) : Nil
```

Comment:
**suggestion:** If the documentation starts with "Removes ..." maybe the method should better be called `#remove` as well?